kpet/data.py

```diff
@@ -56,39 +57,71 @@ def __init__(self, schema, data):
 
 
 class Target:  # pylint: disable=too-few-public-methods, too-many-arguments
-    """Execution target that suite/case patterns match against"""
+    """
+    Execution target which suite/case patterns match against.
+
+    A target has a fixed collection of parameters, each of which can be
+    assigned a target set.
+
+    A target set is either:
+
+        - a set of strings,
+        - Target.ALL, meaning a set containing all possible strings, or
+        - Target.ANY, meaning any possible set of strings.
+    """
+
+    # Empty target set
+    NONE = set()
+    # Complete target set
+    ALL = ()
+    # Any target set
+    ANY = None
+
+    @staticmethod
+    def set_is_valid(target_set):
+        """
+        Check if a target set is valid.
+
+        Args:
+            target_set: The target set to check.
+
+        Returns:
+            True if the target set is valid, false otherwise.
+        """
+        return target_set == Target.ANY or \
+            target_set == Target.ALL or \
+            (isinstance(target_set, set) and
+             set(filter(lambda x: isinstance(x, str), target_set)) ==
+             target_set)
```

Comment:
The last comparison can be written in a slightly more compact (and I'd argue clearer) way:
```
all([isinstance(x, str) for x in target_set])
```

Reply:
Good catch! And the square brackets are not needed either :)

Reply:
Fixed and pushed!